vite/vite.cjs

```diff
@@ -0,0 +1,53 @@
+const { spawnSync, spawn } = require("node:child_process");
+const { parsePayload } = require("./util.cjs");
+
+exports.cliFilename = () => `${__dirname}/vite-cli.mjs`;
+
+exports.build = (options) => {
+  const args = [
+    options.cliFilename,
+    "build",
+    "--options",
+    JSON.stringify(options),
+  ];
+
+  const buffer = spawnSync("node", args, {
+    cwd: options.root,
+    env: {
+      HOME: options.homeEnv,
+      PATH: options.pathEnv,
+    },
+  });
+  const output = buffer.stdout.toString().split("\n");
+  console.log(output.slice(1).join("\n"));
+  return parsePayload(output[0]).outDir;
+};
+
+exports.dev = async (options) => {
+  const args = [
+    options.cliFilename,
+    "dev",
+    "--options",
+    JSON.stringify(options),
+  ];
+
+  const child = spawn("node", args, {
+    cwd: options.root,
+    env: {
+      HOME: options.homeEnv,
+      PATH: options.pathEnv,
+    },
+  });
+
+  return new Promise((resolve) => {
+    child.stdout.on("data", (chunk) => {
+      const payload = parsePayload(chunk.toString());
+      if (payload) {
+        resolve(payload.url);
+        child.stdout.removeAllListeners("data");
+      } else {
+        // console.log(chunk.toString());
```

Comment:
do we need it?

Reply:
Not really, although I think we should have a way to log whatever Vite is outputting somewhere. I had an issue in Wing to discuss this, somewhere. I'll look for it.